src/2025h1/compiletest-directive-rework.md

```diff
@@ -0,0 +1,148 @@
+   `compile-flags`, fix paths related to `aux-build`, etc.
+5. Test writers and reviewers can receive **better diagnostics**, for things like a directive is not
+   accepted in a given test suite or *why* something in `compiletest` failed.
+
+### The "shiny future" we are working towards
+
+My long-term goal for [`compiletest`] is that I want it to make it significantly easier to
+maintain. Concretely, this means significantly better test coverage, easier to extend, better
+documentation. Hopefully, by being more maintainable, we are able to attract more active maintainers
+from both bootstrap and compiler teams and make the code base significantly more pleasant to work
+on.
+
+For directive handling *specifically*, it should mean that:
+
+- It's relatively straightforward and low friction to implement new directives, including test
+  coverage and documentation. It should be easy to do the right thing.
+- [`compiletest`] should produce error messages that are easy to read and understand, possibly even
+  making suggestions.
+- Directives should be documented (and enforced to be documented) via rustdoc which are made
+  available on nightly-rustc docs so we can back-link from dev-guide and not have to maintain two
+  sets of docs that are mutually inconsistent.
+
+## Ownership and team asks
+
+**Owner:** [@jieyouxu]
+
+<!--
+*This section defines the specific work items that are planned and who is expected to do them. It
+should also include what will be needed from Rust teams. The table below shows some common sets of
+asks and work, but feel free to adjust it as needed. Every row in the table should either correspond
+to something done by a contributor or something asked of a team. For items done by a contributor,
+list the contributor, or ![Help wanted][] if you don't yet know who will do it. For things asked of
+teams, list ![Team][] and the name of the team. The things typically asked of teams are defined in
+the [Definitions](#definitions) section below.*
+-->
+
+Note that [`compiletest`] is (in theory) currently co-maintained by both t-bootstrap and t-compiler,
+but AFAIK is (in practice) currently not really actively maintained by anyone else. The following
+team asks are probably mostly [compiler] for feedback on their use cases (as a test infra consumer)
+and [bootstrap] for implementation review.
+
+| Subgoal                                              | Owner(s) or team(s)                         | Notes                                                                                          |
+|------------------------------------------------------|---------------------------------------------|------------------------------------------------------------------------------------------------|
+| General discussion and moral support                 | ![Team][] [bootstrap], ![Team][] [compiler] |                                                                                                |
```

Comment:
Not sure what the syntax is for more than one team there

```text
![Team][] [bootstrap], ![Team][] [compiler]
```

Reply:
```suggestion
| General discussion and moral support                 | ![Team][] [bootstrap], [compiler] |                                                                                                |
```